### ISSUE TITLE
Will pytorch support *Restricted boltzmann machine* basic layer in the future?

### ISSUE THREAD

**Issue by Whisht** [pytorch/pytorch#21137]
## 🚀 Feature

RBM basic layer as `nn.Linear` suppoorted. 
## Motivation

Boltzmann machine is another fundamental generative model in deep learning, But becuase it's object function is computation complicated because of integral of all possible state. So one cannot perform  parameters update through `loss.backward()` and `criterion.step()` as other basic layer in  pytorch. Finally the reconsitution which inherit from `nn.Module` dissatisfy the rbm's needs.

With pytorch, you extend an basic layer from `nn.Module` only needing override the `__init__()` as well as `forward()`. But rbm whose propagation is bidirectional and parameters update usually using **CDk**. For pytorch, the Parameter update process is seperate from the basic layer to `Optimizer`.

So I wonder whether pytorch will make `rbm` as a basic layer and whether will realize `CDk` as a basic optimizer?

Thanks.

**Comment by Whisht** [comment]
In my opinion, for that pytorch is a library for neural network. RBM as a **Graphical Models** probably should not be supported by pytorch.
